resolver.go

```diff
@@ -65,34 +67,45 @@ func (c *resolver) ResolveNow(grpcresolver.ResolveNowOptions) {
 		} else {
 			c.logger.Errorln(err.Error())
 		}
-		c.cc.ReportError(err)
-		return
+		return nil, err
 	}
 
 	addrs := make([]grpcresolver.Address, len(output.Instances))
 	for i, instance := range output.Instances {
 		addrs[i] = httpInstanceSummaryToAddr(instance)
 	}
 
-	c.cc.UpdateState(grpcresolver.State{Addresses: addrs})
+	return &grpcresolver.State{Addresses: addrs}, nil
 }
 
 func (c *resolver) Close() {
-	c.mu.Lock()
-	defer c.mu.Unlock()
-
-	if c.isClosed {
-		return
-	}
-
-	c.isClosed = true
+	c.cancel()
 	c.ticker.Stop()
+	c.wg.Wait()
 }
 
-func (c *resolver) watch() {
+func (c *resolver) watcher() {
+	defer c.wg.Done()
+
 	for {
-		c.ResolveNow(grpcresolver.ResolveNowOptions{})
-		<-c.ticker.C
+		state, err := c.cloudmapLookup()
+		if err != nil {
+			c.cc.ReportError(err)
+		} else {
+			err = c.cc.UpdateState(*state)
+		}
+
+		if err != nil {
+			c.logger.Errorln(err.Error())
```

Comment:
I just find that [grpclog.LoggerV2.Errorln](https://pkg.go.dev/google.golang.org/grpc/grpclog#LoggerV2) handles args like `fmt.Println`.
So, I think we can delete `.Error()` here.

```suggestion
			c.logger.Errorln(err)
```